contracts/utils/convex/ConvexStakingWrapper.sol

```diff
@@ -100,23 +99,40 @@ contract ConvexStakingWrapper is ERC20 {
     function addRewards() public {
         address mainPool = convexPool;
 
-        uint256 rewardsLength = rewards.length;
-
-        if (rewardsLength == 0) {
+        if (rewards.length == 0) {
             RewardType storage reward = rewards.push();
             reward.reward_token = crv;
             reward.reward_pool = mainPool;
-            rewardsLength = 1;
+            reward.reward_integral = 0;
+            reward.reward_remaining = 0;
+
+            reward = rewards.push();
+            reward.reward_token = cvx;
+            reward.reward_pool = address(0); // This is set to address(0) as initially we don't know if the pool has cvx rewards
+            reward.reward_integral = 0;
+            reward.reward_remaining = 0;
+
+            registeredRewards[crv] = CRV_INDEX + 1; //mark registered at index+1
+            registeredRewards[cvx] = CVX_INDEX + 1; //mark registered at index+1
```

Comment:
Why +1? Can't you register them at the index, and then subtract 1 to rewards.length when adding marks for other tokens?